application/tests/unit/common/models/MethodTest.php

```diff
@@ -57,5 +60,133 @@ public function testGetMaskedValueEmail()
         $this->assertEquals('e**************9@d*****.o**', $method->getMaskedValue());
     }
 
+    public function testRuleValidateValueAsPhone()
+    {
+        $method = $this->methods('method1');
+        $this->assertTrue($method->validate(['value']));
+
+        $method->value = 'email@domain.com';
+        $this->assertFalse($method->validate(['value']));
+    }
+
+    public function testCreateAndSendVerificationEmail()
+    {
+        EmailUtils::removeEmailFiles();
+        $user = $this->users('user1');
+
+        $this->assertEquals(0, EmailUtils::getEmailFilesCount());
+
+        $method = Method::createAndSendVerification(
+            $user->id,
+            Method::TYPE_EMAIL,
+            'unique-1461443608@email.com'
+        );
+
+        $this->assertEquals(1, EmailUtils::getEmailFilesCount());
+        $this->assertTrue(EmailUtils::hasEmailFileBeenCreated($method->verification_code));
+        $this->assertTrue(EmailUtils::hasEmailFileBeenCreated($method->value));
+
+    }
+
+    public function testCreateAndSendVerificationPhone()
+    {
+        $user = $this->users('user1');
+        $mockPhones = include __DIR__ . '/../../../mock/phone/data.php';
+        $method = Method::createAndSendVerification(
+            $user->id,
+            Method::TYPE_PHONE,
+            $mockPhones[0]['number']
+        );
+
+        $this->assertEquals($mockPhones[0]['code'], $method->verification_code);
+    }
+
+    public function testCreateAndSendVerificationInvalidType()
+    {
+        $this->setExpectedException('\Exception', '', 1461375342);
+        Method::createAndSendVerification(
+            1,
+            'invalid type',
+            'value'
+        );
+    }
+
+    public function testCreateAndSendVerificationInvalidEmail()
+    {
+        $this->setExpectedException('\Exception', '', 1461459797);
+        Method::createAndSendVerification(
+            1,
+            Method::TYPE_EMAIL,
+            'not-a-email'
+        );
+    }
+
+    public function testCreateAndSendVerificationInvalidPhone()
+    {
+        $this->setExpectedException('\Exception', '', 1461375342);
+        Method::createAndSendVerification(
+            1,
+            Method::TYPE_PHONE,
+            'not-a-phone'
+        );
+    }
+
+    public function testValidateAndSetAsVerifiedValid()
+    {
+        $user = $this->users('user1');
+        $mockPhones = include __DIR__ . '/../../../mock/phone/data.php';
+        $method = Method::createAndSendVerification(
+            $user->id,
+            Method::TYPE_PHONE,
+            $mockPhones[0]['number']
+        );
+
+        $this->assertEquals($mockPhones[0]['code'], $method->verification_code);
```

Comment:
Forgive my ignorance here, but, similar to above (I think), why would the `verification_code` of the `Method` returned by `Method::createAndSendVerification(...)` match the `'code'` in the mock phone data?